src/types/timestamp.rs

```diff
@@ -572,6 +572,66 @@ impl Timestamp {
     pub fn precision(&self) -> Precision {
         self.precision
     }
+
+    /// Returns the year that has been specified in the [Timestamp].
+    pub fn year(&self) -> i32 {
+        self.date_time.year()
+    }
+
+    /// Returns the month that has been specified in the [Timestamp].
+    pub fn month(&self) -> u32 {
+        self.date_time.month()
+    }
+
+    /// Returns the day that has been specified in the [Timestamp].
+    pub fn day(&self) -> u32 {
+        self.date_time.day()
+    }
+
+    /// Returns the hour(s) that has been specified in the [Timestamp].
+    pub fn hour(&self) -> u32 {
```

Comment:
Have you thought about returning an `Option` from these functions so that they can reflect the precision of the `timestamp`?
E.g.
```rust
/// Returns the hour(s) that has been specified in the [Timestamp] if the timestamp's precision 
/// includes hours. If you need hour, regardless of precision, use `hour().unwrap_or_default()`,
/// which will correctly return an hour of `0` when the precision does not include hours.
pub fn hour(&self) -> Option<u32> {
    // ...
}
```

I think it makes it match the data model a little better. It's a little more verbose when you always want some value, but it's a little more ergonomic when you only want to deal with the subfields that are included by the timestamp's precision. What do you think?

Reply:
This is a good observation. I'm unsure which way to go on it; Ion `Timestamp`s _always_ have a hour/minutes/second in that they represent a fixed point in time and those fields are `0` when the precision is too low. However, this does represent an opportunity to remind users that the `0` might have been implicit rather than explicitly part of the value.

I suspect that as a generalized datetime value, providing field defaults is probably what users would prefer. @jpschorr, what's your take on this? Would you rather check the timestamp's `precision`, or `hour().is_none()`?

Reply:
+ 1. I think it would rather simple to check `precision` instead of `hour().is_none()`.